### ISSUE TITLE
logs returned by model.fitDataset onEpochEnd got updated by later epoch

### ISSUE THREAD

**Issue by kangyizhang** [tensorflow/tfjs#984]
The logs returned by `onEpochEnd` during `model.fitDataset` will be override by a later epoch. For example:

const trainLogs:Array<{}> = [];

   const history =
           await model.fitDataset(dataset, {batchesPerEpoch, epochs, callbacks:
            {
            onEpochEnd: async (epoch, logs) => {
              trainLogs.push(logs);
              console.log(trainLogs);
            }
          }
        }); 

for first epoch it would print something like : 

[{"loss":0.9236488342285156}]

for second epoch it would print:

[{"loss":0.7229933540026346},{"loss":0.7229933540026346}]

For model.fit it would be [{"loss":0.9236488342285156},{"loss":0.7229933540026346}]

This can be reproduced by updating any `train_dataset_test` with above code snippet.
